src/main/java/dinkplugin/notifiers/GroupStorageNotifier.java

```diff
@@ -0,0 +1,192 @@
+package dinkplugin.notifiers;
+
+import dinkplugin.message.Field;
+import dinkplugin.message.NotificationBody;
+import dinkplugin.message.NotificationType;
+import dinkplugin.notifiers.data.GroupStorageNotificationData;
+import dinkplugin.notifiers.data.SerializedItemStack;
+import dinkplugin.util.ItemUtils;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.ItemID;
+import net.runelite.api.events.WidgetClosed;
+import net.runelite.api.events.WidgetLoaded;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetID;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.game.ItemManager;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+import java.util.stream.Collectors;
+
+@Singleton
+public class GroupStorageNotifier extends BaseNotifier {
+    static final @VisibleForTesting int GROUP_STORAGE_LOADER_ID = 293;
+    static final @VisibleForTesting String EMPTY_TRANSACTION = "N/A";
+    private static final BinaryOperator<Integer> SUM;
+
+    @Inject
+    private ClientThread clientThread;
+
+    @Inject
+    private ItemManager itemManager;
+
+    private Map<Integer, Integer> initial = Collections.emptyMap();
+
+    @Override
+    public boolean isEnabled() {
+        return config.notifyBank() && super.isEnabled();
+    }
+
+    @Override
+    protected String getWebhookUrl() {
+        return config.bankWebhook();
+    }
+
+    public void onWidgetLoad(WidgetLoaded event) {
+        if (event.getGroupId() != WidgetID.GROUP_STORAGE_GROUP_ID)
+            return;
+
+        clientThread.invokeLater(() -> {
+            ItemContainer inv = getInventory();
+            if (inv == null)
+                return false;
+
+            initial = reduce(inv.getItems());
+            return true;
+        });
+    }
+
+    public void onWidgetClose(WidgetClosed event) {
+        if (event.getGroupId() != GROUP_STORAGE_LOADER_ID)
+            return;
+
+        Widget widget = client.getWidget(GROUP_STORAGE_LOADER_ID, 1);
+        if (widget == null)
+            return;
+
+        if (isEnabled() && StringUtils.containsIgnoreCase(widget.getText(), "Saving")) {
```

Comment:
The current implementation every save operation is successful... unsure what a failure would look like

Reply:
Let's cross that bridge when a report comes in, rather than hunt for it :+1: